### PR TITLE
[Scripts] Updated clean-script

### PR DESCRIPTION
**Pull request by KenAJoh**
### Description

Script is now ESM (.mjs)

Now just removes every `dist` and `.next`-dir found. Node_modules added to fast-glob ignorelist, so no risk of "breaking" dependencies.

Added support for the `--reset` flag that now deletes every node_modules dir if you really want to do a full reset.

### Component Checklist 📝

- [ ] JSDoc
- [ ] Examples
- [ ] Documentation
- [ ] Storybook
- [ ] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [ ] Component tokens (`@navikt/core/css/tokens.json`)
- [ ] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [ ] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [ ] New component? CSS import (`@navikt/core/css/index.css`)
- [ ] Breaking change? Update migration guide. Consider codemod.
- [ ] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

File: scripts/clean.mjs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  });
+  folders
+    .filter((folder) => !folder.includes("node_modules"))
+    .forEach((folder) => {
+      console.info(`Deleting folder ${folder}`);
+      deleteFolder(folder);
+    });
+}
+
+console.info(`Everything looks squeaky clean 🧼`);
```
Comment (HalvorHaugan) [inline scripts/clean.mjs]: Maybe this line is redundant now 🤔

File: scripts/clean.mjs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+console.group("Cleaning up build artifacts");
+
+for (const globPattern of globPatterns) {
+  const folders = FastGlob.sync(globPattern, {
+    dot: true,
+    onlyDirectories: true,
+    ignore: ["**/node_modules"],
+  });
+  folders
+    .filter((folder) => !folder.includes("node_modules"))
```
Comment (HalvorHaugan) [inline scripts/clean.mjs]: Is this needed in addition to `ignore: ["**/node_modules"]`?

File: scripts/clean.mjs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * - Removes all node_modules folders if `--reset`-argv is passed
+ */
+const globPatterns = [
+  "./@navikt/**/lib",
+  "./@navikt/**/esm",
+  "./@navikt/**/cjs",
+  "./@navikt/aksel-icons/src",
+  "**/playwright-report",
+  "**/.next",
+  "**/dist",
```
Comment (HalvorHaugan) [inline scripts/clean.mjs]: I think we should have `"**/dist"` first. Now it first deletes `@navikt/aksel-icons/dist/react/cjs` and then `@navikt/aksel-icons/dist`...

File: scripts/clean.mjs
Comment (HalvorHaugan) [inline scripts/clean.mjs:1]: What are the benefits of ESM?
Should we convert to TS as well?

File: scripts/clean.mjs (hunk 2 above)
Comment (KenAJoh) [reply scripts/clean.mjs]: Nope

File: scripts/clean.mjs
Comment (KenAJoh) [reply scripts/clean.mjs:1]: Benefits are just more modern syntax. 
As for TS, if we require node 22 when its LTS, we can convert it to TS and run 
```
node --experimental-strip-types ...
```
while using built-in nodejs glob. We can then run the script without needing node_modules

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**